### PR TITLE
improve docs for BalanceLaws

### PR DESCRIPTION
**Pull request by simonbyrne**
# Description

Hopefully should make this clearer.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by blallen** [comment]
this is all API style documentation, right?

**Comment by simonbyrne** [comment]
> this is all API style documentation, right?

Yes.

File: src/BalanceLaws/interface.jl [hunk 1]
```diff
@@ -1,22 +1,62 @@
...
+- [`vars_state`](@ref) to define the prognostic, auxiliary and intermediate
+  variables.
+- [`flux_first_order!`](@ref) to compute ``F_1``
+- [`flux_second_order!`](@ref) to compute ``F_2``
+- [`source!`](@ref) to compute ``S``
+
+If `vars(bl, ::GradientFlux, FT)` is non-empty, then the following should be
+defined:
+- [`compute_gradient_argument!`](@ref) to compute ``G``
+- [`compute_gradient_flux!`](@ref) is a linear transformation of ``\nabla g``
```
Comment (charleskawczynski) [inline src/BalanceLaws/interface.jl:27]:
```suggestion
- [`compute_gradient_flux!`](@ref) is a linear transformation of ``∇g``
```
At the moment, `\n` is parsed as a newline.

**Review by charleskawczynski — APPROVED** [review]
I added one syntax fix. Aside from that, this looks like a nice set of improvements.

File: src/BalanceLaws/interface.jl (hunk 1 above)
Comment (simonbyrne) [reply src/BalanceLaws/interface.jl:27]: Good catch. Thanks!

**Pull request merged**